examples_tests/61.BlitFilterTest/main.cpp

```diff
@@ -0,0 +1,1561 @@
+			cmdbuf->pipelineBarrier(asset::EPSF_COMPUTE_SHADER_BIT, asset::EPSF_COMPUTE_SHADER_BIT, asset::EDF_NONE, 0u, nullptr, 1u, &alphaTestBarrier, 1u, &readyForNorm);
+
+			cmdbuf->bindDescriptorSets(asset::EPBP_COMPUTE, normalizationPipeline->getLayout(), 0u, 1u, &normalizationDS);
+			cmdbuf->bindComputePipeline(normalizationPipeline);
+			CBlitFilter::normalization_push_constants_t normPC = {};
+			CBlitFilter::dispatch_info_t normDispatchInfo = {};
+			buildNormalizationParameters(inImageExtent, outImageExtent, referenceAlpha, normPC, normDispatchInfo);
+			dispatchHelper<CBlitFilter::normalization_push_constants_t>(cmdbuf, normalizationPipeline->getLayout(), normPC, normDispatchInfo);
+		}
+	}
+
+	//! WARNING: This function blocks and stalls the GPU!
+	void blit(
+		video::IGPUQueue* computeQueue,
+		const BlitFilter::CState::E_ALPHA_SEMANTIC alphaSemantic,
+		video::IGPUDescriptorSet* alphaTestDS,
+		video::IGPUComputePipeline* alphaTestPipeline,
+		video::IGPUDescriptorSet* blitDS,
+		video::IGPUComputePipeline* blitPipeline,
+		video::IGPUDescriptorSet* normalizationDS,
+		video::IGPUComputePipeline* normalizationPipeline,
+		const core::vectorSIMDu32& inImageExtent,
+		const asset::IImage::E_TYPE inImageType,
+		const asset::E_FORMAT inImageFormat,
+		core::smart_refctd_ptr<video::IGPUImage> outImage,
+		const float referenceAlpha = 0.f,
+		const asset::SBufferRange<video::IGPUBuffer>* alphaTestCounter = nullptr)
+	{
+		auto cmdpool = device->createCommandPool(computeQueue->getFamilyIndex(), video::IGPUCommandPool::ECF_NONE);
+		core::smart_refctd_ptr<video::IGPUCommandBuffer> cmdbuf;
+		device->createCommandBuffers(cmdpool.get(), video::IGPUCommandBuffer::EL_PRIMARY, 1u, &cmdbuf);
+
+		auto fence = device->createFence(video::IGPUFence::ECF_UNSIGNALED);
+
+		cmdbuf->begin(video::IGPUCommandBuffer::EU_ONE_TIME_SUBMIT_BIT);
+		blit(cmdbuf.get(), alphaSemantic, alphaTestDS, alphaTestPipeline, blitDS, blitPipeline, normalizationDS, normalizationPipeline, inImageExtent, inImageType, inImageFormat, outImage, referenceAlpha, alphaTestCounter);
+		cmdbuf->end();
+
+		video::IGPUQueue::SSubmitInfo submitInfo = {};
+		submitInfo.commandBufferCount = 1u;
+		submitInfo.commandBuffers = &cmdbuf.get();
+		computeQueue->submit(1u, &submitInfo, fence.get());
+
+		device->blockForFences(1u, &fence.get());
+	}
+
+	inline asset::E_FORMAT getOutImageViewFormat(const asset::E_FORMAT format)
+	{
+		const auto& formatUsages = device->getPhysicalDevice()->getImageFormatUsagesOptimal(format);
+
+		if (formatUsages.storageImage)
+		{
+			return format;
+		}
+		else
+		{
+			const asset::E_FORMAT compatFormat = getCompatClassFormat(format);
+
+			const auto& compatClassFormatUsages = device->getPhysicalDevice()->getImageFormatUsagesOptimal(compatFormat);
+			if (!compatClassFormatUsages.storageImage)
+				return asset::EF_UNKNOWN;
+			else
+				return compatFormat;
+		}
+	}
+
+	static inline const char* getGLSLFormatStringFromFormat(const asset::E_FORMAT format)
```

Comment:
hoist to `IGLSLCompiler` and call it `getStorageImageFormatQualifier`

Reply:
TODO

Reply:
Resolved.